tests/test_end_to_end.py

```diff
@@ -53,49 +54,100 @@ def nbstripout_exe():
 
 
 @pytest.mark.parametrize("input_file, expected_file, args", TEST_CASES)
-def test_end_to_end_stdin(input_file: str, expected_file: str, args: List[str]):
+@pytest.mark.parametrize("verify", (True, False))
+def test_end_to_end_stdin(input_file: str, expected_file: str, args: List[str], verify: bool):
     with open(NOTEBOOKS_FOLDER / expected_file, mode="r") as f:
         expected = f.read()
+        expected_str = json.dumps(json.loads(expected), indent=2)
```

Comment:
Same question: why do you need to convert to JSON?

Reply:
Inside the test my main concern was that line endings might be different between the input and the output files. (If my memory is not failing me...)

Although to be fair this might have been an overstep. since it actually changes the test ... it tests whether the content is equivalent instead of testing that the file is actually the same. (lmk if I should revert it here)

Reply:
This should indeed not be required. When I investigated this I was going crazy cause there was a persistently failing test, until I realized this was due to some extra leading whitespace in one of the test notebook input files, which I fixed in 5077ab4. Can you please rebase your PR on main and then make those changes?

Reply:
thanks for the fix, will do.

Reply:
Sorry I missed this one in the tests (focused on changing the string serializaition in the actual runtime code)

Reply:
Done in [4babd2a](https://github.com/kynan/nbstripout/pull/195/commits/4babd2a3099c7968f3e713cfe4eb6cbbdc9728c4)